pkg/vm/stackitem/item.go

```diff
@@ -0,0 +1,990 @@
+func (i *Interop) String() string {
+	return "Interop"
+}
+
+// Dup implements Item interface.
+func (i *Interop) Dup() Item {
+	// reference type
+	return i
+}
+
+// Bool implements Item interface.
+func (i *Interop) Bool() bool { return true }
+
+// TryBytes implements Item interface.
+func (i *Interop) TryBytes() ([]byte, error) {
+	return nil, errors.New("can't convert Interop to ByteArray")
+}
+
+// TryInteger implements Item interface.
+func (i *Interop) TryInteger() (*big.Int, error) {
+	return nil, errors.New("can't convert Interop to Integer")
+}
+
+// Equals implements Item interface.
+func (i *Interop) Equals(s Item) bool {
+	if i == s {
+		return true
+	} else if s == nil {
+		return false
+	}
+	val, ok := s.(*Interop)
+	return ok && i.value == val.value
+}
+
+// Type implements Item interface.
+func (i *Interop) Type() Type { return InteropT }
+
+// Convert implements Item interface.
+func (i *Interop) Convert(typ Type) (Item, error) {
+	switch typ {
+	case InteropT:
+		return i, nil
+	case BooleanT:
+		return NewBool(i.Bool()), nil
+	default:
+		return nil, errInvalidConversion
+	}
+}
+
+// MarshalJSON implements the json.Marshaler interface.
+func (i *Interop) MarshalJSON() ([]byte, error) {
+	return json.Marshal(i.value)
+}
+
+// Pointer represents VM-level instruction pointer.
+type Pointer struct {
+	pos    int
+	script []byte
+	hash   util.Uint160
+}
+
+// NewPointer returns new pointer on the specified position.
+func NewPointer(pos int, script []byte) *Pointer {
+	return &Pointer{
+		pos:    pos,
+		script: script,
+		hash:   hash.Hash160(script),
+	}
+}
+
+// String implements Item interface.
+func (p *Pointer) String() string {
+	return "Pointer"
+}
+
+// Value implements Item interface.
+func (p *Pointer) Value() interface{} {
+	return p.pos
+}
+
+// Dup implements Item interface.
+func (p *Pointer) Dup() Item {
+	return &Pointer{
+		pos:    p.pos,
+		script: p.script,
+		hash:   p.hash,
+	}
+}
+
+// Bool implements Item interface.
+func (p *Pointer) Bool() bool {
+	return true
+}
+
+// TryBytes implements Item interface.
+func (p *Pointer) TryBytes() ([]byte, error) {
+	return nil, errors.New("can't convert Pointer to ByteArray")
+}
+
+// TryInteger implements Item interface.
+func (p *Pointer) TryInteger() (*big.Int, error) {
+	return nil, errors.New("can't convert Pointer to Integer")
+}
+
+// Equals implements Item interface.
+func (p *Pointer) Equals(s Item) bool {
+	if p == s {
+		return true
+	}
+	ptr, ok := s.(*Pointer)
+	return ok && p.pos == ptr.pos && p.hash == ptr.hash
+}
+
+// Type implements Item interface.
+func (p *Pointer) Type() Type {
+	return PointerT
+}
+
+// Convert implements Item interface.
+func (p *Pointer) Convert(typ Type) (Item, error) {
+	switch typ {
+	case PointerT:
+		return p, nil
+	case BooleanT:
+		return NewBool(p.Bool()), nil
+	default:
+		return nil, errInvalidConversion
+	}
+}
+
+// Hash returns pointer item hash
+func (p *Pointer) Hash() util.Uint160 {
```

Comment:
IMO `ScriptHash()` is more appropriate for the external interface: we can have multiple pointers with the same `hash`, while every other `Hash()` across the codebase has different signature and implies uniqueness.